### ISSUE TITLE
Using FP8 mixed-precision in accelerate give error. Where and how do I install the necessary packages?

### ISSUE THREAD

**Issue by Signorlimone** [kohya-ss/sd-scripts#1553]
As per title, I'm experimenting with 8gb gpu training and would like to use fp8 precision for speed improvement but I have this error:

ValueError: Using `fp8` precision requires `transformer_engine` or `MS-AMP` to be installed.

Is there a way to install this packages?

**Comment by b-7777777** [comment]
chatgpt:
The error you're encountering suggests that your code or model is attempting to use FP8 (8-bit floating point) precision, which requires additional libraries such as transformer_engine or MS-AMP (Microsoft Automatic Mixed Precision) to be installed.

To resolve this issue, you have two main options:

1. Install transformer_engine:
transformer_engine is a library provided by NVIDIA that supports FP8 precision for training and inference on NVIDIA GPUs.

You can install it using pip:
pip install transformer-engine

2. Install Microsoft AMP (MS-AMP):
If you prefer to use Microsoft's Automatic Mixed Precision, you can install it. This is more commonly used in the PyTorch ecosystem.

To install MS-AMP, you might need to install torch and torchvision with the appropriate version:

pip install torch torchvision torchaudio
pip install ms-amp

**Comment by b-7777777** [comment]
Does pip installing the missing libraries not resolve the issue?

**Comment by Signorlimone** [comment]
> fp8 is already used in FLUX.1 LoRA training. transformer_engine is included in Accelerate. Therefore, it would be quite difficult to train with 8GB.

true, but I was able to train with bf16 precision and it's really slow. fp16 is way much better but it gives avr_loss=nan. That's why I wanted to try fp8 precision and it gave me the error where it asks to install MS-AMP

**Comment by Tam678** [comment]
> > fp8 is already used in FLUX.1 LoRA training. transformer_engine is included in Accelerate. Therefore, it would be quite difficult to train with 8GB.
> 
> true, but I was able to train with bf16 precision and it's really slow. fp16 is way much better but it gives avr_loss=nan. That's why I wanted to try fp8 precision and it gave me the error where it asks to install MS-AMP

same problem,have you solved the problem？

**Comment by sbhavani** [comment]
@Tam678 could you share a repro? I'd be happy to look into it. Using FP8 with accelerate should work as it automatically replaces gthe linear layers (with fp8 layers)
